java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JCRObject.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.java.generator.nodes;
+
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.expr.Name;
+import com.github.javaparser.ast.expr.SingleMemberAnnotationExpr;
+import com.github.javaparser.ast.expr.StringLiteralExpr;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JCRObject extends AbstractJSONSchema2Pojo {
+
+    private final String type;
+    private final String group;
+    private final String version;
+    private final boolean withSpec;
+    private final boolean withStatus;
+
+    public JCRObject(
+            String type, String group, String version, boolean withSpec, boolean withStatus) {
+        this.type = type;
+        this.group = group;
+        this.version = version;
+        this.withSpec = withSpec;
+        this.withStatus = withStatus;
+    }
+
+    @Override
+    public String getType() {
+        return this.type;
+    }
+
+    @Override
+    public GeneratorResult generateJava(CompilationUnit cu) {
+        ClassOrInterfaceDeclaration clz =
+                cu.getClassByName(this.type).orElse(cu.addClass(this.type));
+
+        clz.addAnnotation(
+                new SingleMemberAnnotationExpr(
+                        new Name("io.fabric8.kubernetes.model.annotation.Version"),
+                        new StringLiteralExpr(version)));
+        clz.addAnnotation(
+                new SingleMemberAnnotationExpr(
+                        new Name("io.fabric8.kubernetes.model.annotation.Group"),
```

Comment:
If these classes are on the classpath, I'd recommend using `Class.getName()` methods

Reply:
Reasons against string literals 
- Typos are easy
- Model could change overtime

Reply:
`Class.getName()` would pick up the name of the class during the `runtime` of the generator and prevent it's usage e.g. as a shaded dependency.

Regarding the raised points:
- typos -> we have compilation tests
- if the model changes it will break the test suite

Reply:
> if the model changes it will break the test suite

Shouldn't that be expected behavior? Otherwise, the generated classes themselves would fail to compile, no?

Reply:
Yup, that's expected, but the compilation unit of this codebase is not the same as the user's, also, this is the very same approach used in the CRD generator codebase.